src/helper.cc

```diff
@@ -1560,23 +1564,22 @@ Helper::Session::checkForTimedOutRequests(bool const retry)
         requestsIndex.erase(it);
         requests.pop_front();
         debugs(84, 2, "Request " << r->request.Id << " timed-out, remove it from queue");
-        void *cbdata;
         bool retried = false;
         if (retry && r->request.retries < MAX_RETRIES && cbdataReferenceValid(r->request.data)) {
             debugs(84, 2, "Retry request " << r->request.Id);
             ++r->request.retries;
             parent->submitRequest(r);
             retried = true;
-        } else if (cbdataReferenceValidDone(r->request.data, &cbdata)) {
+        } else if (cbdataReferenceValid(r->request.data)) {
```

Comment:
This is another cbdataReferenceValid() left in the code, for very similar reasons.

The two cbdataReferenceValid() calls in this code can be merged into one by introducing a higher level `if` statement, but that would create a lot of PR noise, so this optimization is best avoided in this PR IMO.